File: lihua-biz/lihua-file/src/main/java/com/lihua/entity/SysAttachment.java
package com.lihua.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.lihua.model.validation.AttachmentValidation;
import com.lihua.mybatis.model.BaseEntity;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serial;
import java.io.Serializable;

@EqualsAndHashCode(callSuper = true)
@Data
@Accessors(chain = true)
public class SysAttachment extends BaseEntity implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 主键
     */
    @TableId
    private String id;

    /**
     * 附件存储名
     */
    private String storageName;

    /**
     * 附件原名称
     */
    private String originalName;

    /**
     * 附件扩展名
     */
    private String extensionName;

    /**
     * 附件保存路径
     */
    private String path;

    /**
     * 分片上传id
     */
    @NotNull(message = "uploadId为空", groups = { AttachmentValidation.AttachmentChunksMergeUploadValidation.class })
    private String uploadId;

    /**
     * 业务编码（默认附件上传时所在的路由名称）
     */
    private String businessCode;

    /**
     * 业务名称（默认附件上传时所在的菜单名称）
     */
    private String businessName;

    /**
     * 附件大小
     */
    private String size;

    /**
     * 附件类型
     */
    private String type;

    /**
     * 上传方式 一般上传、分片上传、附件秒传
     */
    private String uploadMode;

    /**
     * 状态 上传成功、上传失败、分片上传中、业务删除
     */
    private String status;

    /**
     * 附件存储位置 如：本地、云存储等
     */
    private String storageLocation;

    /**
     * md5值
     */
    @NotNull(message = "md5值不能为空", groups = { AttachmentValidation.AttachmentCheckMd5Validation.class })
    private String md5;

    /**
     * 上传失败原因
     */
    private String errorMsg;

    /**
     * 原url（通过url上传有该字段）
     */
    @NotNull(message = "原URL为空", groups = { AttachmentValidation.AttachmentUrlUploadValidation.class })
    private String url;

    /**
     * 上传客户端类型
     */
    private String clientType;
}

File: lihua-app/src/api/system/attachment/type/sys-attachment.ts
export interface SysAttachment {
    /** 主键 */
    id?: string;
    /** 文件存储名 */
    storageName?: string;
    /** 文件原名称 */
    originalName?: string;
    /** 文件扩展名 */
    extensionName?: string;
    /** 文件保存路径 */
    path?: string;
    /** 分片上传id */
    uploadId?: string;
    /** 业务编码（默认文件上传时所在的路由名称） */
    businessCode?: string;
    /** 业务名称（默认文件上传时所在的菜单名称） */
    businessName?: string;
    /** 文件大小 */
    size?: string;
    /** 文件类型 */
    type?: string;
    /** 上传方式 */
    uploadMode?: string;
    /** 上传状态 上传成功、上传失败 */
    status?: string;
    /** 文件存储位置 如：本地、云存储等 */
    storageLocation?: string;
    /** md5 */
    md5?: string;
    /** 上传人id */
    createId?: string;
    /** 上传时间 */
    createTime?: Date;
    /** 删除标识 */
    delFlag?: string;
    /** 上传失败原因 */
    errorMsg?: string;
    /** 原url（通过url上传有该字段） */
    url?: string;
}

export interface SysAttachmentDTO extends SysAttachment {

    /**
     * 创建时间集合
     */
    createTimeList?: Date[];

    /**
     * 当前页数
     */
    pageNum: number;

    /**
     * 每页记录数
     */
    pageSize: number;

}

export interface SysAttachmentVO extends SysAttachment {
    /**
     * 上传用户昵称
     */
    uploadName?: string;
}